### ISSUE TITLE
Azure not being run after feedstock creation

### ISSUE THREAD

**Issue by dpryan79** [conda-forge/conda-smithy#1154]
For the last two feedstocks I created via staged-recipes, only Appveyor was run on the feedstock after its creation. That's a problem as the recipes were `noarch: generic`, meaning that it did nothing. After noticing this, I added a single space to the `meta.yaml` to trigger the CI again and then the normal Azure processes were triggered and the packages actually built. The two feedstocks that I know were affected by this are:

 - [r-seqminer](https://github.com/conda-forge/r-seqminer-feedstock)
 - [r-fateid](https://github.com/conda-forge/r-fateid-feedstock)

**Comment by scopatz** [comment]
This feedstock got three commits, though CI was run on the second: https://github.com/conda-forge/simpleitk-feedstock/commits/master

**Comment by scopatz** [comment]
The waiting was implemented properly ([See example here](https://travis-ci.org/conda-forge/staged-recipes/builds/588984435#L883)) but it didn't seem to work

**Comment by scopatz** [comment]
And only appveyor ran on this feedstock: https://github.com/conda-forge/wisdem-feedstock/commits/master

**Comment by jakirkham** [comment]
@dpryan79, does that answer your question?

**Comment by dpryan79** [comment]
@jakirkham If https://github.com/conda-forge/staged-recipes/pull/9640 solved this then yes. I haven't created any feedstocks in the past few days, so I can't personally confirm if this is resolved or not.

**Comment by scopatz** [comment]
This issue is not resolved

**Comment by npavlovikj** [comment]
I second @scopatz , I am having the same issue with my feedstock created yesterday.

**Comment by jakirkham** [comment]
I've emailed some people who work on Azure about it. Maybe they have some ideas. 🙂

**Comment by leantk** [comment]
Hi all, I'm Leah and my team owns the GitHub and Azure Pipelines integration and I was pointed to help out with this issue. Could you point me to some specific commits that didn't start a pipeline? I see some different repos mentioned above but I want to make sure I'm digging into the right commit. I wasn't sure which commits had a pipeline created on our end already or when setup was still taking place. Sorry for the inconvenience :-(

**Comment by jakirkham** [comment]
Hi Leah 👋 Thanks for coming by.

Recently we saw this with commit ( https://github.com/conda-forge/ogre-feedstock/commit/e96f50b94a633309f610eafa1f189dc10764bd12 ). So manually started it. This issue is primarily discussing that problem. Namely a feedstock (repo) is freshly created and registered on Azure. Once we perform the first push of these two commits, Azure doesn't appear to run.

However with this specific feedstock, it seems a recent PR ( https://github.com/conda-forge/ogre-feedstock/pull/2 ) is having the same issue. Not sure if that is a sign of the same underlying issue or not.

Should be able to point you to some fresh commits in the morning, which may prove more useful still.

Thanks for your help. 🙂

**Comment by leantk** [comment]
Ok so I looked into https://github.com/conda-forge/ogre-feedstock/commit/e96f50b94a633309f610eafa1f189dc10764bd12 not firing a pipeline and what I see on our end is that the event dropped because it said the pipeline didn't exist. From what I can tell, the pipeline was created seconds before the CI event came to us so it looks like a race condition. Is your process to create a pipeline and do this initial commit for CI?

**Comment by isuruf** [comment]
Yes, we added a 10 second wait. We can increase the waiting time. How many seconds/minutes is reasonable?

**Comment by jakirkham** [comment]
@leantk, here's a fresh one ( https://github.com/conda-forge/pyuoi-feedstock/issues/1 ).

**Comment by leantk** [comment]
I would say try 120 seconds to play it safe if that works for you. This isn't something I've seen before but it looks like for these 2 instances the definition creation time and the time the CI event came in are 30-40 seconds apart. We can manually kickoff builds immediately without a delay so the logic around CI events trying to find the definition must not be getting the most up-to-date information which I'll dig into some more. For now, if you are ok with increasing the timeout, I think that gets you a workaround at least so you aren't seeing builds not running every so often. Thanks letting us know about this and sorry for the trouble
